include/leveldb/expiry.h

```diff
@@ -0,0 +1,98 @@
+// -------------------------------------------------------------------
+//
+// expiry.h:  background expiry management for Basho's modified leveldb
+//
+// Copyright (c) 2016 Basho Technologies, Inc. All Rights Reserved.
+//
+// This file is provided to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file
+// except in compliance with the License.  You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+// -------------------------------------------------------------------
+
+#ifndef EXPIRY_H
+#define EXPIRY_H
+
+#include <stdint.h>
+#include "leveldb/env.h"
+#include "util/refobject_base.h"
+
+namespace leveldb {
+
+class Compaction;
+class Logger;
+class ParsedInternalKey;
+class Slice;
+class SstCounters;
+class Version;
+class VersionEdit;
+struct FileMetaData;
+
+class ExpiryModule : public RefObjectBase
+{
+public:
+    ExpiryModule() {};
+    ~ExpiryModule() {};
```

Comment:
The dtor is implicitly virtual since the dtor in `RefObjectBase` base class is declared as virtual, but I still think it's helpful to declare it explicitly as virtual in derived classes.